File: tv-fork-patches/patched/common/src/main/java/com/liskovsoft/smartyoutubetv2/common/litetube/LitetubePrefs.java
package com.liskovsoft.smartyoutubetv2.common.litetube;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * Persistent Litetube state: client JWT, cached proxy pool, activation code.
 */
public class LitetubePrefs {
    private static final String PREFS = "litetube_prefs";
    private static final String KEY_JWT = "jwt";
    private static final String KEY_ACTIVATION_CODE = "activation_code";
    private static final String KEY_CACHED_POOL = "cached_proxy_pool";
    private static final String KEY_API_BASE = "api_base";

    public static final String DEFAULT_API_BASE = "https://api.litetube.trfnv.ru";

    private static volatile LitetubePrefs instance;
    private final SharedPreferences sp;

    private LitetubePrefs(SharedPreferences sp) {
        this.sp = sp;
    }

    public static LitetubePrefs instance(Context context) {
        if (instance == null) {
            synchronized (LitetubePrefs.class) {
                if (instance == null) {
                    instance = new LitetubePrefs(
                            context.getApplicationContext()
                                    .getSharedPreferences(PREFS, Context.MODE_PRIVATE));
                }
            }
        }
        return instance;
    }

    public boolean hasValidJwt() {
        String jwt = getJwt();
        return jwt != null && !jwt.isEmpty();
    }

    public String getJwt() {
        return sp.getString(KEY_JWT, null);
    }

    public void setJwt(String jwt) {
        sp.edit().putString(KEY_JWT, jwt).apply();
    }

    public void clearJwt() {
        sp.edit().remove(KEY_JWT).apply();
    }

    public String getActivationCode() {
        return sp.getString(KEY_ACTIVATION_CODE, null);
    }

    public void setActivationCode(String code) {
        sp.edit().putString(KEY_ACTIVATION_CODE, code).apply();
    }

    public void clearActivationCode() {
        sp.edit().remove(KEY_ACTIVATION_CODE).apply();
    }

    public void setCachedProxyPoolJson(String json) {
        sp.edit().putString(KEY_CACHED_POOL, json).apply();
    }

    public String getCachedProxyPoolJson() {
        return sp.getString(KEY_CACHED_POOL, null);
    }

    public String getLitetubeApiBase() {
        String base = sp.getString(KEY_API_BASE, DEFAULT_API_BASE);
        return base != null ? base : DEFAULT_API_BASE;
    }

    public void setLitetubeApiBase(String base) {
        sp.edit().putString(KEY_API_BASE, base.trim().replaceAll("/$", "")).apply();
    }
}

File: tv-fork-patches/patched/common/src/main/java/com/liskovsoft/smartyoutubetv2/common/litetube/LitetubePrefs.kt
package com.liskovsoft.smartyoutubetv2.common.litetube

import android.content.Context
import android.content.SharedPreferences

/**
 * Persistent Litetube state: client JWT (post-activation), cached proxy
 * pool (last /api/proxy/pool response), and the activation code currently
 * being advertised on TV.
 *
 * Plain SharedPreferences: Android TV apps are sandboxed and rooted-TVs are
 * a separate hardening phase. Anything sensitive (JWT) gets revoked server-side
 * by /api/auth/* anyway, so on-device encryption adds little.
 */
class LitetubePrefs private constructor(private val sp: SharedPreferences) {

    fun hasValidJwt(): Boolean = !getJwt().isNullOrBlank()

    fun getJwt(): String? {
        val raw = sp.getString(KEY_JWT, null) ?: return null
        // Stored value is the raw JWT (no Bearer prefix); LitetubeApi.baseRequest
        // adds the Authorization header. Treat any non-blank value as valid -
        // server-side validation on next API call will reject forged tokens.
        return raw.takeIf { it.isNotBlank() }
    }

    fun setJwt(jwt: String) {
        sp.edit().putString(KEY_JWT, jwt).apply()
    }

    fun clearJwt() {
        sp.edit().remove(KEY_JWT).apply()
    }

    fun getActivationCode(): String? = sp.getString(KEY_ACTIVATION_CODE, null)

    fun setActivationCode(code: String) {
        sp.edit().putString(KEY_ACTIVATION_CODE, code).apply()
    }

    fun clearActivationCode() {
        sp.edit().remove(KEY_ACTIVATION_CODE).apply()
    }

    fun setCachedProxyPoolJson(json: String) {
        sp.edit().putString(KEY_CACHED_POOL, json).apply()
    }

    fun getCachedProxyPoolJson(): String? = sp.getString(KEY_CACHED_POOL, null)

    fun getLitetubeApiBase(): String =
        sp.getString(KEY_API_BASE, DEFAULT_API_BASE) ?: DEFAULT_API_BASE

    fun setLitetubeApiBase(base: String) {
        sp.edit().putString(KEY_API_BASE, base.trimEnd('/')).apply()
    }

    companion object {
        private const val PREFS = "litetube_prefs"
        private const val KEY_JWT = "jwt"
        private const val KEY_ACTIVATION_CODE = "activation_code"
        private const val KEY_CACHED_POOL = "cached_proxy_pool"
        private const val KEY_API_BASE = "api_base"

        // Default points at the public Litetube gateway; mirrors the upstream
        // upstream SmartTube build targets test servers by default.
        const val DEFAULT_API_BASE = "https://api.litetube.trfnv.ru"

        @Volatile private var instance: LitetubePrefs? = null

        @JvmStatic
        fun instance(context: Context): LitetubePrefs =
            instance ?: synchronized(this) {
                instance ?: LitetubePrefs(
                    context.applicationContext.getSharedPreferences(PREFS, Context.MODE_PRIVATE)
                ).also { instance = it }
            }
    }
}
